libs/build_netcdf.sh

```diff
@@ -63,7 +63,7 @@ LDFLAGS1="-L$HDF5_ROOT/lib"
 LDFLAGS2=$(cat $HDF5_ROOT/lib/libhdf5.settings | grep AM_LDFLAGS | cut -d: -f2)
 [[ $enable_pnetcdf =~ [yYtT] ]] && LDFLAGS4="-L$PNETCDF_ROOT/lib"
 if [[ ${STACK_netcdf_shared:-} != [yYtT] ]]; then
-  LDFLAGS1+=" -lhdf5_hl -lhdf5"
+  LDFLAGS1+=" -lhdf5 -lhdf5_hl -lz"
```

Comment:
why `lz`? Isn't that captured in L67 with the `Extra libraries`?
Also, what if hdf5 was built without zlib?

Reply:
This is now reverted to the develop branch of noaa-emc/hpc-stack